File: benchmark/c/cpw/position_of_rightmost_set_bit_1/position_of_rightmost_set_bit_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int position_of_rightmost_set_bit_1 ( int n ) ;
int position_of_rightmost_set_bit_1 ( int n ) {
  int position = 1;
  int m = 1;
  while ( ! ( n & m ) ) {
    m = m << 1;
    position ++;
  }
  return position;
}

File: translation_result/else_bms/fail/c/cpw/position_of_rightmost_set_bit_1/gpt4/Restart/position_of_rightmost_set_bit_1.rs

fn position_of_rightmost_set_bit_1(n: i32) -> i32 {
    let mut position = 1;
    let mut m = 1;
    while (n & m) == 0 {
        m = m.wrapping_shl(1);
        position += 1;
    }
    position
}
